### PR TITLE
Add a optional pass to lower cases

### PR DESCRIPTION
**Pull request by kripken**
This is useful for comparing switches with and without cases. It also turns out to be simple to do, a nice example of the pass system I think.

### CODE REVIEW

File: src/passes/LowerCase.cpp [hunk 1]
```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2015 WebAssembly Community Group participants
```
Comment (jfbastien) [inline src/passes/LowerCase.cpp]: 2016

Comment (kripken) [inline src/passes/LowerCase.cpp]: :( I keep copy-pasting. Should we update all the existing ones too?

Comment (jfbastien) [inline src/passes/LowerCase.cpp]: The guidance changes on this, but the latest I've gotten is that the file date should be the one that the file was created on (it used to be: update every time a substantial modification happens, with date ranges).

This should all get fixed when / if we re-license :-)

Comment (kripken) [inline src/passes/LowerCase.cpp]: Ok. I fixed this one meanwhile.

**Pull request merged**